docs/design-proposals/2022-04 TraceQL Concepts.md

```diff
@@ -0,0 +1,198 @@
+
+## Structure
+
+A query is an expression that is evaluated on one trace at a time, and iterates, evaluating each trace. The expression is evaluated on each trace as a whole, one at a time. If the evaluation produces a set of spans, then the trace and the set of spans is part of the result set.
+
+The structure of a query is an expression or set of chained expressions (a pipeline). Chained expressions are evaluated left to right.  Each expression selects or discards this trace from being included in the results set.
+
+## Selecting spans
+
+In TraceQL the curly brackets `{}` always select a set of spans from the current trace. They will commonly be paired with a condition to reduce the spans being passed in.
+
+`{ http.status = 200 }`
+
+This simple query will be evaluated on every span of every trace, one at a time. 
+
+If the trace being evaluated contains *no* spans with an attribute`http.status` with the value `200` then no spans will be selected and this trace will not appear in our resultset.
+If the trace does contain spans with an attribute`http.status` with the value `200` then *only* those spans will be returned. The trace is reduced to only the set of spans that match the condition inside the `{}`. This set of spans (and therefore trace) will be included in our resultset.
+
+### Intrinsic fields
+
+Every span has certain intrinsic fields that can be referenced when selecting spans
+
+| field name | description |
+| ---------- | ----------- |
+| `start`    | start time of the span |
+| `end`      | end time of the span |
+| `duration` | end - start time of the span |
+| `name`     | operation or span name |
+| `status`   | status values are "error", "ok", or "unset" |
+| `parent`   | the parent of this span | 
+
+**Examples**
+Find traces that contain spans whose duration is greater than 2 seconds:  
+`{ duration > 2s }`
+
+Find traces that contain a span named "HTTP POST":  
+`{ name = "HTTP POST"  }`
+
+Find traces that cross service boundaries:  
+`{ parent.service.name != service.name }`
+
+### Attribute fields
+
+We can refer to dynamic attributes (also known as tags) on the span or the span's resource.
+
+**Examples**
+
+Find traces with the GET http method:  
+`{ http.method = "GET" }`
+
+Find traces that passed through the prod namespace:  
+`{ namespace = "prod" }`
+
+### Field expressions
+
+Fields can also be combined in various expected ways.
+
+**Examples**
+
+Find traces where the difference of the duration of a set of parent/child spans exceeds 500ms:  
+`{ parent.duration - duration > 500ms }`
+
+Find traces with "success" http status codes:  
+`{ http.status >= 200 && http.status < 300 }`
+
+Note that the second expression requires both conditions to be true on the same span. The entire expression inside of `{}` must be evaluated as true on a single span for it to be included in the resultset.
+
+## Combining Spansets
```

Comment:
maybe add a section introducing the result set data model and define `spanset` there

Reply:
Good point. I am going to reword the "structure" section above to better highlight how integral spansets are to the language and add some details about the resultset.